src/OrchardCore/OrchardCore.ResourceManagement/TagHelpers/ScriptTagHelper.cs

```diff
@@ -158,6 +158,47 @@ public override async Task ProcessAsync(TagHelperContext context, TagHelperOutpu
 
             var childContent = await output.GetChildContentAsync();
 
+            if (!string.IsNullOrEmpty(DependsOn))
+            {
+                var dependencies = DependsOn.Split(_separator, StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries);
+
+                foreach (var dependency in dependencies)
+                {
+                    var dependencyParts = dependency.Split(';');
```

Comment:
Is the semicolon used somewhere already? I understand it as being a way to specify location. However, I don't think we need that: if the dependency is already required, then it'll have a location specified. Otherwise, the dependency should follow the location of its dependent resource, i.e. use `At`.

Reply:
> Is the semicolon used somewhere already?
No.

>  If the dependency is already required, then it'll have a location specified.

I don't think we have a default placement for resources. So somehow we have to determine a location for the dependency. In this case, we default to the value in  `At` but add the flexibility to specify an explicit location.

Reply:
I don't want to introduce new logic if possible.


In both cases I expect that test would be injected before the inline script. If that is that case without using the logic I added, then we should be able to remove it. I'll test it out tomorrow 

<script at="Head" depends-on="test"></script>

<script at="Foot" depends-on="test"></script>

Reply:
@Piedone I updated the logic and removed the `;location` logic from the tag helpers.